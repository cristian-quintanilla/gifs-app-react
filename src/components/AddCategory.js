import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');
	
	const handleInputChange = e => {
		setInputValue(e.target.value);
		// console.log('handleInputChange para pruebas');
	}

	const handleSubmit = e => {
		e.preventDefault();
		// console.log('handleSubmit para pruebas');
		
		// Validación
		if (inputValue.trim().length > 2){
			setCategories(categories => [ inputValue, ...categories ]);
			setInputValue('');
		}
	}

	return (
		<form onSubmit={ handleSubmit } >
			<p>{ inputValue }</p>
			<input
				type="text"
				placeholder="Agregar Categoría..."
				className="input--style-6"
				value={ inputValue }
				onChange={ handleInputChange } />
		</form>
	)
}

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired
}