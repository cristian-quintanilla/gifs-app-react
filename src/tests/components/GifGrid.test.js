import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el Componente <GifGrid />', () => {
	
	const category = 'Jujutsu Kaisen';

	test('debe de mostrar el componente correctamente', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true
		});

		const wrapper = shallow( <GifGrid category={ category } /> );
		expect( wrapper ).toMatchSnapshot();
	});

	test('debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {
		const gifs = [
			{
				id: 1,
				url: 'https://localhost/api/image_1.jpg',
				title: 'Cualquier Cosa'
			}, {
				id: 2,
				url: 'https://localhost/api/image_2.jpg',
				title: 'Cualquier Cosa 2'
			}
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false
		});

		const wrapper = shallow( <GifGrid category={ category } /> );
		
		expect( wrapper ).toMatchSnapshot();
		expect( wrapper.find('p').exists() ).toBe( false );
		expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
	});

});