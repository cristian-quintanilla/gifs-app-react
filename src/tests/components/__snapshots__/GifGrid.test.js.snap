// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Componente <GifGrid /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Jujutsu Kaisen
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el Componente <GifGrid /> debe de mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Jujutsu Kaisen
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={1}
      key="1"
      title="Cualquier Cosa"
      url="https://localhost/api/image_1.jpg"
    />
    <GifGridItem
      id={2}
      key="2"
      title="Cualquier Cosa 2"
      url="https://localhost/api/image_2.jpg"
    />
  </div>
</Fragment>
`;
